Dockerfile

```diff
@@ -1,49 +1,59 @@
-FROM node:10.3-alpine
-LABEL maintainer="https://github.com/verdaccio/verdaccio"
+FROM node:10.3-alpine as builder
 
-RUN apk --no-cache add openssl && \
-    wget -O /usr/local/bin/dumb-init https://github.com/Yelp/dumb-init/releases/download/v1.2.0/dumb-init_1.2.0_amd64 && \
-    chmod +x /usr/local/bin/dumb-init && \
-    apk del openssl && \
-    apk --no-cache add ca-certificates wget && \
+RUN apk --no-cache add openssl ca-certificates wget && \
     wget -q -O /etc/apk/keys/sgerrand.rsa.pub https://raw.githubusercontent.com/sgerrand/alpine-pkg-glibc/master/sgerrand.rsa.pub && \
-    wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.25-r0/glibc-2.25-r0.apk && \
+    wget -q https://github.com/sgerrand/alpine-pkg-glibc/releases/download/2.25-r0/glibc-2.25-r0.apk && \
     apk add glibc-2.25-r0.apk
 
-ENV APPDIR /usr/local/app
-
-WORKDIR $APPDIR
-
-ADD . $APPDIR
+WORKDIR /opt/verdaccio-build
+COPY . .
 
-ENV NODE_ENV=production
+ENV NODE_ENV=production \
+    VERDACCIO_BUILD_REGISTRY=https://registry.npmjs.org/
 
-RUN npm config set registry http://registry.npmjs.org/ && \
-    yarn global add -s flow-bin@0.69.0 && \
+RUN yarn config set registry $VERDACCIO_BUILD_REGISTRY && \
     yarn install --production=false && \
     yarn lint && \
     yarn code:docker-build && \
     yarn build:webui && \
     yarn cache clean && \
     yarn install --production=true --pure-lockfile
 
+
+
+FROM node:10.3-alpine
+LABEL maintainer="https://github.com/verdaccio/verdaccio"
+
+RUN apk --no-cache add openssl dumb-init
+
 RUN mkdir -p /verdaccio/storage /verdaccio/plugins /verdaccio/conf
 
+ENV VERDACCIO_APPDIR=/opt/verdaccio
+WORKDIR $VERDACCIO_APPDIR
+
+COPY --from=builder /opt/verdaccio-build .
+
 ADD conf/docker.yaml /verdaccio/conf/config.yaml
 
-RUN addgroup -S verdaccio && adduser -S -G verdaccio verdaccio && \
-    chown -R verdaccio:verdaccio "$APPDIR" && \
-    chown -R verdaccio:verdaccio /verdaccio
+ENV PATH=${VERDACCIO_APPDIR}/bin:${PATH} \
+    HOME=${VERDACCIO_APPDIR} \
+    VERDACCIO_USER_NAME=verdaccio \
+    VERDACCIO_USER_UID=10001
+
+RUN adduser -u ${VERDACCIO_USER_UID} -S -D -h ${VERDACCIO_APPDIR} -g "${VERDACCIO_USER_NAME} user" -s /sbin/nologin ${VERDACCIO_USER_NAME} && \
+    chmod -R +x ${VERDACCIO_APPDIR}/bin && \
+    chown -R ${VERDACCIO_USER_UID}:root /verdaccio/storage && \
+    chmod -R g=u /verdaccio/storage /etc/passwd
 
-USER verdaccio
+USER $VERDACCIO_USER_UID
 
-ENV PORT 4873
-ENV PROTOCOL http
+ENV VERDACCIO_PORT 4873
+ENV VERDACCIO_PROTOCOL http
 
-EXPOSE $PORT
+EXPOSE $VERDACCIO_PORT
 
-VOLUME ["/verdaccio"]
+VOLUME /verdaccio/storage
 
-ENTRYPOINT ["/usr/local/bin/dumb-init", "--"]
+ENTRYPOINT ["uid_entrypoint"]
 
-CMD $APPDIR/bin/verdaccio --config /verdaccio/conf/config.yaml --listen $PROTOCOL://0.0.0.0:${PORT}
+CMD $VERDACCIO_APPDIR/bin/verdaccio --config /verdaccio/conf/config.yaml --listen $VERDACCIO_PROTOCOL://0.0.0.0:${VERDACCIO_PORT}
```

Comment:
I see some access to env vars with $NAME and ${NAME}, can we unify that?

Reply:
Just updated that in the last push. It was leftovers from tests.